### PR TITLE
General: move VaultPress class to own file & add PHP53 shortcircuit

### LINKED ISSUE #15: PHP 5.2: Fatal Error
#5 introduced new PHP 5.3+ code to VaultPress, and as a result you now get a Fatal Error if you use the plugin on a site using an old version of PHP.

```
Warning: Unexpected character in input: '\' (ASCII=92) state=1 in /wp-content/plugins/vaultpress/vaultpress.php on line 701
Parse error: syntax error, unexpected T_STRING in /wp-content/plugins/vaultpress/vaultpress.php on line 701
```

We would need to bail early if you use an old version of PHP on your site.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #15

- Moving the main VaultPress class to its own class allows us to move all PHP 5.3+ code out of the plugin's entry point, vaultpress.php
- In that plugin's entry point, we now check the PHP version in use on the site, and pause the VaultPress plugin (i.e. stop loading it) if you are running an old version of PHP.

image

**To test**

- Try activating the plugin on a site running PHP 5.2. It should be paused.
- Try activating the plugin on a site running a more modern version of PHP; nothing should change.

### CODE REVIEW

File: vaultpress.php [hunk 1]
```diff
@@ -9,18 +9,69 @@
...
+	/**
+	 * Outputs an admin notice for folks running an outdated version of PHP.
+	 *
+	 * @todo: Remove once WP 5.2 is the minimum version.
+	 *
+	 * @since 2.0.0
+	 */
+	function vaultpress_admin_unsupported_php_notice() { ?>
+		<div class="notice notice-error is-dismissible">
+			<p><?php esc_html_e( 'VaultPress requires a more recent version of PHP and has been paused. Please update PHP to continue enjoying VaultPress.', 'vaultpress' ); ?></p>
```
Comment (dereksmart) [inline vaultpress.php]: I like how the error_log above actually includes version numbers in the message. Can we add those here?

File: vaultpress.php [hunk 2]
```diff
@@ -9,18 +9,69 @@
...
+			<p class="button-container">
+				<?php
+				printf(
+					'<a class="button button-primary" href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a>',
+					esc_url( wp_get_update_php_url() ),
```
Comment (dereksmart) [inline vaultpress.php]: Could `wp_get_update_php_url()` fatal a pre-5.1 WP install?

Comment (jeherve) [reply vaultpress.php]: Yes, that's a good point. I assume we don't want to have VaultPress require WP 5.1 just yet.

b2b2a71d5119d24cdf8a08d6f56a24be0dc96c15 should solve this issue.

File: vaultpress.php (hunk 1 above)
Comment (jeherve) [reply vaultpress.php]: Sounds good. Done in 4b59173c469f06ca8177345ee3e3b9be5e17e46d

image

**Pull request merged**